src/ML/pytorch/valueMLmodel.cc

```diff
@@ -39,47 +40,21 @@ void valueMLmodel::learn(RLDataset& dataSet, unsigned int numberOfEpochs, unsign
 vector<float> valueMLmodel::calcValueEstimates(state* s) {
     /* Calculate all possible successor states of s.
        Returns a boolean vector corresponding to edges in such a way that true means an edge is selectable */
-    vector<bool> index = s->calcSuccessorStates();     
-
-    torch::Tensor succStates = vectorToTensor(s->successorStates);
-
-    // Returns the value prediction for all successor states.       
-    torch::Tensor pred = linearNet->forward(succStates);
-
-    vector<float> tempPrediction = tensorToVector(pred);
-    vector<float> prediction;
-    unsigned int counter = 0;
-
-    // To get a vector of the size of the state, we set 0 for every not selectable edge.
     for (unsigned int i = 0; i < dimstate; i++) {
-        if (index.at(i)) {
-            prediction.push_back(tempPrediction.at(counter));
-            counter++;
-        } else {
-        prediction.push_back(0);
-        }
-    }                                        
-    return prediction;                              
-}
-
-// Transforms a vector of states into a tensor
-torch::Tensor valueMLmodel::vectorToTensor(vector<vector<bool>>& vec) {
-    torch::Tensor tens = torch::zeros({(long int) vec.size(),(long int) vec.at(0).size()});
-    for (unsigned int i = 0; i < vec.size(); i++){
-        for (unsigned int j = 0; j < vec.at(i).size(); j++) {                
-            tens[i][j] = (float)vec.at(i).at(j);
-        }           
-    }
-    return tens;
-}
-
-// Transforms a tensor of values for the successor states into a vector
-vector<float> valueMLmodel::tensorToVector(torch::Tensor& tens) {
-    vector<float*> vec;
-    vector<float> vecReturn;
-    for (unsigned int i = 0; i < tens.numel(); i++) {
-        vec.push_back(tens[i].data<float>());
-        vecReturn.push_back(*vec.at(i));
+        tensState[i] = (float) s->selectedSubset.at(i);
+        
     }
-    return vecReturn;
+    vector<unsigned int> index = s->calcSuccessorStates();     
+    vector<float*> prediction(dimstate,0);
```

Comment:
Something like `vector<float*>` is very uncommon. I'm not quiet sure what you want to achieve but you probably do it wrong this way.